networks/1_correlation_analysis.py

```diff
@@ -16,35 +16,35 @@
 
 # # Correlation analysis
 #
-# This notebook performs correlation analysis to compare the similarity between genes and applies different threshold cutoffs to determine the strength of connection between genes
+# This notebook examines the correlation structure in the gene expression data generated in [1_create_compendia.ipynb](../processing/1_create_compendia.ipynb).
+#
+# When we performed clustering on the correlation matrices (using Pearson correlation) we found that pairs of genes had either very high correlation scores (>0.5) or very low correlation scores (<0.1). As a result gene pairs that were highly correlated clustered into a single large module. This finding is consistent with a [previous study](https://link.springer.com/article/10.1186/1471-2164-7-187), which found that KEGG (a database that containes genes or proteins annotated with specific biological processes as reported in the literature) is bias in some biological processes represented. Figure 1C demonstrates that a large fraction of gene pairs are ribosomal relationships - in the top 0.1% most co-expressed genes, 99% belong to the ribosome pathway.
+# Furthermore, protein function prediction based on co-expression drop dramatically after removing the ribisome pathway (Figure 1A, B).
+#
+# This notebook applies different corrections to try to remove this very dominant global signal in the data. This notebook follows from [1a_transformation_correlation_analysis.ipynb](1a_transformation_correlation_analysis.ipynb). Here we are applying dimensionality reduction techniques in addition to scaling the data.
 
 # %load_ext autoreload
 # %autoreload 2
 import os
 import pandas as pd
 import plotnine as pn
 import seaborn as sns
+from sklearn import preprocessing
 import matplotlib.pyplot as plt
 import umap
 import random
 import numpy as np
-from scipy.spatial.distance import pdist, squareform
+import scipy
+from sklearn.decomposition import PCA
 from core_acc_modules import paths
 
 # ## Set user parameters
 #
-# For now we will vary the correlation threshold (`corr_threshold`) but keep the other parameters consistent
-#
-# We will run this notebook for each threshold parameter
+# Here we set the number of PCs or singular vectors to use. We are starting with 300 since this is what [eADAGE](https://pubmed.ncbi.nlm.nih.gov/28711280/) used.
 
-# +
 # Params
-corr_threshold = 0.5
-
-# Output files
-pao1_membership_filename = f"pao1_membership_{corr_threshold}.tsv"
-pa14_membership_filename = f"pa14_membership_{corr_threshold}.tsv"
-# -
+num_singular_values = 300
```

Comment:
Did you try varying these numbers at all?

Would it be possible to make variance explained plots like people do for PCA, to see how much of the variance in the data you're picking up for various numbers of singular values? It seems like you can do this if you use the [sklearn version of SVD](https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.TruncatedSVD.html), although I'm not sure if the numpy implementation provides this function out of the box.

Reply:
Yes I varied this number a bit, but 300 seemed reasonable given that eADAGE uses 300, though that is a nonlinear model.

So for both SVD and PCA, the variance explained was captured in fewer than 10 PCs and 20 singular vectors, but with such few singular vectors or PCs, there we were still seeing very large modules.

Reply:
These curves should be in [1b_corrected_correlation_analysis.ipynb](networks/archive/1b_corrected_correlation_analysis.ipynb)